### ISSUE TITLE
[BUG]: Stable diffusion output a random noise

### ISSUE THREAD

**Issue by yufengyao-lingoace** [hpcaitech/ColossalAI#2437]
![WX20230111-183123@2x](https://user-images.githubusercontent.com/92853025/211783641-da829483-6437-444f-8ccd-ad3604719670.png)
I tried to train the teyvat data using: python main.py --logdir /tmp/ -t -b configs/Teyvat/train_colossalai_teyvat.yaml. Then i got a ckpt file of 9.7g.
But when i use python scripts/txt2img.py --prompt "a photograph of an astronaut riding a horse" --plms
    --outdir ./output \
    --config path/to/logdir/checkpoints/last.ckpt \
    --ckpt /path/to/logdir/configs/project.yaml  \
I got a picture with random noise. (I just follow the step of this readme file:https://github.com/hpcaitech/ColossalAI/tree/main/examples/images/diffusion)

I think you yourself didn't try yout code, because this code of you is obviously wrong:python scripts/txt2img.py --prompt "a photograph of an astronaut riding a horse" --plms
    --outdir ./output \
    --config path/to/logdir/checkpoints/last.ckpt \
    --ckpt /path/to/logdir/configs/project.yaml  \
the config and ckpt is inversed.

So is it that the ckpt file created by colossal is not right or any other problem?

**Comment by 1SAA** [comment]
Hi @yufengyao-lingoace 

Have you tried the latest ColossalAI? You can install the newest version from source. As for your problem, I'm trying to repeat this bug.

**Comment by yufengyao-lingoace** [comment]
@1SAA Hi,
I tried both pip install colossalai==0.1.12+torch1.12cu11.3 -f https://release.colossalai.org and install from source. But got the same result. The train process is very normal:
Epoch 0:  80%|███▉ | 130/163 [13:22<03:23,  6.17s/it, loss=0.425, v_num=0, train/loss_simple_step=0.334, train/loss_vlb_step=0.334, train/loss_step=0.334, global_step=129.0, lr_abs=1.6e-7]

**Comment by 1SAA** [comment]
Hi @yufengyao-lingoace 

yeah. I got the same random noise. I reckon it's a problem with model saving. I'm trying to figure it out.

**Comment by yufengyao-lingoace** [comment]
@1SAA  ok, thanks very much, looking forward to you!

**Comment by yufengyao-lingoace** [comment]
@1SAA Thanks for your reply.  Is it means that without the pre-trained model, we can not get a good output after a few epoch? I tried to train with 2000 pictures with 60 epoches, but I still get the picture with random noise. Is it because the 60 epoch is not enough for outputing a good picture?

**Comment by QiuLL** [comment]
I tried dreambooth，found random noise. So I tried to save the model in the first epoch eveb without been update. Then i used inference.py to generate picture，and still get wrong results. 
commands:python inference_qll.py --prompt "a girl with red hat" --modelpath "/data/qll/ColossalAI_3/ColossalAI/outputs/checkpoint-0"
in train_dreambooth_colossalai.py:
    torch.cuda.synchronize()
    for epoch in range(args.num_train_epochs):
        unet.train()
        for step, batch in enumerate(train_dataloader):
            if gpc.get_local_rank(ParallelMode.DATA) == 0 and False:
                logger.info(f"before saving and pipline", ranks=[0])
                pipeline = DiffusionPipeline.from_pretrained(
                    args.pretrained_model_name_or_path,
                    unet=convert_to_torch_module(unet),
                    revision=args.revision,
                )
                logger.info(f"after pipline, before saving", ranks=[0])
                save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
                pipeline.save_pretrained(save_path)
                logger.info(f"Saving model checkpoint to {save_path}", ranks=[0])
                .......
in inference.py:
    opt = parser.parse_args()
    # seed_everything(opt.seed)

    model_id = opt.modelpath
    print(f"Loading model... from{model_id}")

    pipe = DiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16).to("cuda")

    pipe.safety_checker = lambda images, **kwargs: (images, False)

    prompt = opt.prompt  # "A photo of an apple."
    image = pipe(prompt, num_inference_steps=50, guidance_scale=7.5).images[0]

    image.save("output.png")

![image](https://user-images.githubusercontent.com/1172763/216009206-41a86247-fc5d-48f3-b570-5c8901566220.png)

**Comment by EricZgw** [comment]
Is this a useless project?

**Comment by Fazziekey** [comment]
> Is this a useless project?

It have been fixed in https://github.com/hpcaitech/ColossalAI/pull/2561

**Comment by Fazziekey** [comment]
Nosie problem by colossal veriosion have been fixed in https://github.com/hpcaitech/ColossalAI/pull/2561
